mergify_engine/clients/github.py

```diff
@@ -112,6 +113,17 @@ def items(self, url, api_version=None, list_items=None, **params):
             else:
                 break
 
+    def request(self, *args, **kwargs):
+        try:
+            return super().request(*args, **kwargs)
+        finally:
+            self._nb_requests += 1
+
+    def close(self):
+        super().close()
```

Comment:
I feel like this is going to be forgotten. Any chance to use this as a context manager?

Reply:
It was ugly with context manager because the get_client also raise MergifyNotInstalled.

I look at it and found a cleaner way to do so.